x/epochs/types/hooks.go

```diff
@@ -0,0 +1,55 @@
+package types
+
+import (
+	"context"
+	"errors"
+
+	"cosmossdk.io/core/appmodule"
+)
+
+type EpochHooks interface {
+	// the first block whose timestamp is after the duration is counted as the end of the epoch
+	AfterEpochEnd(ctx context.Context, epochIdentifier string, epochNumber int64, env appmodule.Environment) error
+	// new epoch is next block of epoch end block
+	BeforeEpochStart(ctx context.Context, epochIdentifier string, epochNumber int64, env appmodule.Environment) error
+	// Returns the name of the module implementing epoch hook.
+	GetModuleName() string
+}
+
+var _ EpochHooks = MultiEpochHooks{}
+
+// combine multiple gamm hooks, all hook functions are run in array sequence.
+type MultiEpochHooks []EpochHooks
+
+// GetModuleName implements EpochHooks.
+func (MultiEpochHooks) GetModuleName() string {
+	return ModuleName
+}
+
+func NewMultiEpochHooks(hooks ...EpochHooks) MultiEpochHooks {
+	return hooks
+}
+
+// AfterEpochEnd is called when epoch is going to be ended, epochNumber is the number of epoch that is ending.
+func (h MultiEpochHooks) AfterEpochEnd(ctx context.Context, epochIdentifier string, epochNumber int64, env appmodule.Environment) error {
```

Comment:
mmmh, shall we not pass the environment here for now, and mention the case here: https://github.com/cosmos/cosmos-sdk/issues/19640.
For consistency, I don't think we should settle for this api yet until we figure #19640 direction
As it would fit in that issue, given that the module defining its hook will get the whole environment of the epoch module.

cc @tac0turtle

Reply:
Removed, also not used for anything except logging

Reply:
yes env would be good here